ssl/record/ssl3_record.c

```diff
@@ -381,9 +389,9 @@ int ssl3_get_record(SSL *s)
         goto f_err;
     }
 #ifdef SSL_DEBUG
-    printf("dec %d\n", rr->length);
+    printf("dec %ld\n", rr->length);
```

Comment:
%ld is not the right specifier for a size_t. %d is a signed type while size_t is unsigned, and a size_t doesn't have to have the same size as a long and doesn't at least on 64 bit windows. In C99 you should use %zu. We probably need something like PRIu32 for platforms that don't support %z.